### ISSUE TITLE
Kestrel returns 415 if Content-Type header isn't provided for GET calls

### ISSUE THREAD

**Issue by Eilon** [aspnet/AspNetCore#5707]
_From @manish-panwar on Tuesday, 18 December 2018 19:21:20_

# Issue Title

Kestrel in net core 2.2 returns 415 if Content-Type header isn't provided for GET call

# General

I am using .net core 2.2 and created a web api. When we make GET call with "Accept" header then Kestrel is  returning 415, it works fine if I pass the Content-Type header. In dotnet core 2.1.x it used to work fine even if I didn't pass Content-Type header. I am wondering what has changed in 2.2

Generally one shouldn't need to pass Content-Type header for GET call, and Accept header should be enough.

I hope this is the right place to raise this question. If not then if someone knows where I can raise this issue. Thanks a lot for your help!

_Copied from original issue: dotnet/core#2166_

**Comment by dacki** [comment]
I am experiencing the same issues when adding `ConsumesAttribute` to MvcOptions

```
services.AddMvc(options =>
{
    ...
    options.Filters.Add(new ConsumesAttribute("application/json"));
});
```

Then basically any request (`GET`, `POST`, `DELETE` tested) without content will fail due to `415 Unsupported` despite not having any content or body specified in the according Controller methods.

Versions used:

- NetCore 2.2
- AspNetCore.HttpOverrides 2.2.0
- AspNetCore.Mvc 2.2.0
- AspNetCore.Routing 2.2.0
- AspNetCore.Server.Kestrel 2.2.0
- AspNetCore.StaticFiles 2.2.0
- AspNetCore.TestHost 2.2.0

**Comment by muratg** [comment]
We believe 415 is set by MVC.

**Comment by shessane** [comment]
Same issue here. It occurs when using the Consumes attribute.

```c#
[Route("api/[controller]")]
[Consumes("application/json")]
[ApiController]
public class ValuesController : ControllerBase
{
	// GET api/values
	[HttpGet]
	public ActionResult<IEnumerable<string>> Get()
	{
		return new string[] { "value1", "value2" };
	}
}
```

Trace logs :
```
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request starting HTTP/1.1 GET http://localhost:49991/api/values  
Microsoft.AspNetCore.HostFiltering.HostFilteringMiddleware:Trace: All hosts are allowed.
Microsoft.AspNetCore.Routing.Matching.DfaMatcher:Debug: 1 candidate(s) found for the request path '/api/values'
Microsoft.AspNetCore.Routing.Matching.DfaMatcher:Debug: Endpoint '415 HTTP Unsupported Media Type' with route pattern '(none)' is valid for the request path '/api/values'
Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware:Debug: Request matched endpoint '415 HTTP Unsupported Media Type'
Microsoft.AspNetCore.Routing.EndpointMiddleware:Information: Executing endpoint '415 HTTP Unsupported Media Type'
Microsoft.AspNetCore.Routing.EndpointMiddleware:Information: Executed endpoint '415 HTTP Unsupported Media Type'
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request finished in 18.0389ms 415 
```

Is it a bug or a breaking change from 2.1 ?

**Comment by shessane** [comment]
It seem's to be a regression. The correction is already merged (#6645), and should be part of the 2.2.2 release.

**Comment by mkArtakMSFT** [comment]
Closing as fixed, per @shessane 's response.
